ocis-pkg/store/store.go

```diff
@@ -0,0 +1,88 @@
+package store
+
+import (
+	"context"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/owncloud/ocis/v2/ocis-pkg/store/etcd"
+	"github.com/owncloud/ocis/v2/ocis-pkg/store/memory"
+	"go-micro.dev/v4/store"
+)
+
+var (
+	storeEnv        = "OCIS_STORE"
+	storeAddressEnv = "OCIS_STORE_ADDRESS"
+	storeOCMemSize  = "OCIS_STORE_OCMEM_SIZE"
+)
+
+var ocMemStore *store.Store
+
+// Get the configured key-value store to be used.
+//
+// Each microservice (or whatever piece is using the store) should use the
+// options available in the interface's operations to choose the right database
+// and table to prevent collisions with other microservices.
+// Recommended approach is to use "services" or "ocis-pkg" for the database,
+// and "services/<service-name>/" or "ocis-pkg/<pkg>/" for the package name.
+//
+// So far, only the name of the store and the node addresses are configurable
+// via environment variables.
+// Available options for "OCIS_STORE" are:
+// * "noop", for a noop store (it does nothing)
+// * "etcd", for etcd
+// * "ocmem", custom in-memory implementation, with fixed size and optimized prefix
+// and suffix search
+// * "memory", for a in-memory implementation, which is the default if noone matches
+//
+// "OCIS_STORE_ADDRESS" is a comma-separated list of nodes that the store
+// will use. This is currently usable only with the etcd implementation. If it
+// isn't provided, "127.0.0.1:2379" will be the only node used.
+//
+// "OCIS_STORE_OCMEM_SIZE" will configure the maximum capacity of the cache for
+// the "ocmem" implementation, in number of items that the cache can hold per table.
+// You can use "OCIS_STORE_OCMEM_SIZE=5000" so the cache will hold up to 5000 elements.
+// The parameter only affects to the "ocmem" implementation, the rest will ignore it.
+// If an invalid value is used, the default will be used instead, so up to 512 elements
+// the cache will hold.
```

Comment:
I think it will depend on the plans we have for the cache.

I assumed that we want to use the same cache for all the services, so each service shouldn't need to configure a cache for itself. In addition, each service won't be able to use a different cache other than the one predefined; for example, ocs won't be able to use an in-memory cache and graph won't be able to use a distributed cache if the "global" cache is set to noop.
Using the same cache also simplifies things because we only need to configure the cache in one place, and the services will only need to get the already-configured instance.

Moving the configuration to the services' configuration implies that each service can configure its own cache. This will add additional complexity and I'm not sure about the advantages. The usual scenarios are: use an in-memory cache for the all-in-one deployment, or use a distributed cache for a distributed deployment. There is no real benefit of an in-memory cache in a distributed deployment unless the cache is owned by the service. For the case of the roles, the cache is owned by the roleManager, which is shared across multiple services.

I agree these options need to be more visible, but attaching them to a service might not be what we want.